### PR TITLE
Don't silently fail when the directory does not exist

### PR DESCRIPTION
**Pull request by Joelasaur**
Ran into an issue where I was trying to publish a package that used `import-modules`, and ran into a fun bug where the directory being read would work fine from inside the module, but not when imported from  a second package. It would've been helpful to get an error describing what happened instead of failing silently - it would've saved me several `npm publish` commands :) 

Anyways I added error throwing and a test to prove that it works.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -22,8 +21,8 @@ module.exports = (directory, options) => {
 	let files;
 	try {
 		files = fs.readdirSync(directory);
-	} catch (_) {
-		return {};
+	} catch (error) {
```
Comment (sindresorhus) [inline index.js]: The try/catch is moot now.

File: index.js [hunk 2]
```diff
@@ -12,7 +12,6 @@ const fileExtensions = new Set(['.js', '.json', '.node']);
 
 module.exports = (directory, options) => {
 	directory = path.resolve(parentDirectory, directory || '');
-
```
Comment (sindresorhus) [inline index.js]: Don't make unrelated changes.

File: test.js [hunk 3]
```diff
@@ -11,5 +11,11 @@ test('main', t => {
 	t.deepEqual(Object.keys(importModules('fixture', {camelize: false, fileExtensions: ['.xjs']})), ['non-standard']);
 	t.deepEqual(Object.keys(importModules()), ['index', 'package']);
 	t.deepEqual(importModules('fixture/empty'), {});
-	t.deepEqual(importModules('non-existent'), {});
+});
+
+test('non-existent directory', t => {
+	const error = t.throws(() => {
```
Comment (sindresorhus) [inline test.js]: Prefer using the `t.throws` native error message matching (you can use a regex)

File: index.js (hunk 2 above)
Comment (Joelasaur) [reply index.js]: ah didn't realize I did this, thanks

File: index.js (hunk 1 above)
Comment (Joelasaur) [reply index.js]: Ok, removed it and did this instead
```javascript
	const files = fs.readdirSync(directory);
```

File: test.js (hunk 3 above)
Comment (Joelasaur) [reply test.js]: I looked into regex, but `String` has `includes` built in so why not use it? Can `includes` be used in `t.throws` as a matcher?

Comment (Joelasaur) [reply test.js]: Nevermind, I just got the regex working. Pushing now.

**Comment by Joelasaur** [comment]
@sindresorhus Is this able to go in?
